### ISSUE TITLE
Re-assess old download endpoints and services

### ISSUE THREAD

**Issue by terrywbrady** [CDLUC3/mrt-doc#351]
Once the presigned URL's are fully in place, assess if any old endpoints for downloads should be retired.
- What about OpenContext?
- What about UNM storage if applicable
- What about Pairtree use for testing purposes

Services
- How much traffic is Express receiving?

**Comment by mreyescdl** [comment]
EScholarship seems to be the last client using Merritt Express (see sample request below)
If we are able to transition them to pre-signed, then we should be able to shutdown Express.

Let's meet with Martin et. al. to see if this could happen.

`GET /wsgi-bin/download.wsgi?container=uc3-s3mrt5001-prd&cloudType=aws&node=5001&file=ark:/13030/qt4jw8q8nw%7C24%7Cproducer/meta/qt4jw8q8nw.meta.xml `

**Comment by terrywbrady** [comment]
@elopatin-uc3 , do you want to take the action item above to set up a follow up with the eScholarship team?  Since they are only grabbing a file the switch over may be very simple.

**Comment by elopatin-uc3** [comment]
@terrywbrady yes, I'll reach out to them to mention this, and then plan a meeting for us to sync up.

**Comment by elopatin-uc3** [comment]
@terrywbrady @mreyescdl – Martin mentions the above request is used only to check for the existence of a file. He's suggested using an API call instead, but I'm not sure if we have an appropriate endpoint.

**Comment by terrywbrady** [comment]
If there is not already a way to do this in the UI, let's add it!

**Comment by elopatin-uc3** [comment]
It turns out that they are actually downloading the file and checking its contents. 

Martin shared this code sample:
`# Now form a URL to get the metadata file in Merritt.
    # Switching to MrtExpress since it's faster
    mrtMetaURL = "https://mrtexpress.cdlib.org/dl/ark:/13030/%s/meta/%s.meta.xml" % (shortArk, shortArk)
    #mrtMetaURL = "https://merritt.cdlib.org/d/%s/0/producer%%2Fmeta%%2F%s.meta.xml" % (escapedArk, shortArk)`

He will be out next week, but said Brian or Mahjabeen could make the change to a presigned URL format.

**Comment by elopatin-uc3** [comment]
Sample to suggest, including python string formatting:
`https://merritt.cdlib.org/api/presign-file/%s/0/producer%%2Fmeta%%2F%s.meta.xml" % (escapedArk, shortArk)`

**Comment by elopatin-uc3** [comment]
Mahjabeen will be filing a ticket in the eSchol pivotal project (I do not have access to this currently, but Justin should be help with that): https://www.pivotaltracker.com/n/projects/1320348

**Comment by elopatin-uc3** [comment]
@mreyescdl, Mahjabeen has confirmed that the above eScholarship requests have been modified to use the presign-file endpoint. We should no longer see this traffic hitting Express.

**Comment by mreyescdl** [comment]
@elopatin-uc3 Thanks Eric,
The traffic from the harvester stopped around 1:00 today.
We'll need to archive the source code and submit to Merritt.  Then decommission.

https://github.com/CDLUC3/mrt-doc/issues/396
